apps/framework-cli/src/cli.rs

```diff
@@ -1008,21 +1008,85 @@ pub async fn cli_run() {
 
     info!("CLI Configuration loaded and logging setup: {:?}", config);
 
-    let cli = Cli::parse();
+    // Setup Ctrl+C / SIGINT handler
+    tokio::spawn(async move {
+        // Set up a handler for Ctrl+C / SIGINT
+        match tokio::signal::ctrl_c().await {
+            Ok(()) => {
+                info!("Received Ctrl+C / SIGINT, initiating graceful shutdown");
+
+                // Before anything else, disable any background Redis tasks
+                info!("Stopping background tasks");
+
+                // Read settings directly here to avoid borrowing issues
+                if let Ok(project) = Project::load_from_current_dir() {
+                    if !project.is_production {
+                        let should_skip_shutdown =
+                            match std::env::var("MOOSE_SKIP_CONTAINER_SHUTDOWN") {
```

Comment:
That probably should be part of the project config otherwise we will have environment variables reading sprinkled in the code

Reply:
Make sense - cleaning that up in the next pr update